packages/gatsby-plugin-page-creator/src/__tests__/path-utils.ts

```diff
@@ -0,0 +1,16 @@
+import { pagesGlob } from "../path-utils"
+import micromatch from "micromatch"
+
+describe(`pagesGlob`, () => {
+  it(`will match files with collection routes in any segment of their file path.`, async () => {
+    const pathPatterns = [
+      `{a.slug}.tsx`,
+      `a/{b.slug}.tsx`,
+      `a/{b.slug}/c.tsx`,
+      `a/{b.slug}/c/d.tsx`,
+    ]
+    expect(micromatch(pathPatterns, pagesGlob)).toHaveLength(
+      pathPatterns.length
+    )
+  })
+})
```

Comment:
This unit test runs the pattern against a list using the library underlying `globby`- is this enough or do we want to mock the filesystem and use `globby` itself? My opinion is that this is good enough for the logic we want to test, but there's good arguments on either side so I'm happy to defer to maintainers' opinion.

Reply:
`globby` could in theory switch its implementation under the hood in the future (e.g. switching to a Rust NAPI implementation to speed things up) and then that test would be irrelevant but still pass.

So we'll want to test `findCollectionPageFiles` and mock out anything that's necessary :)